src/daria/analysis/concentrationanalysis.py

```diff
@@ -216,24 +254,33 @@ def find_cleaning_filter(
             baseline_images (list of daria.Image): series of baseline_images.
             reset (bool): flag whether the cleaning filter shall be reset.
         """
+        print("Setting up a cleaning filter.")
+
         # Initialize cleaning filter
         if reset:
-            self.threshold = np.zeros_like(self.base, dtype=float)
+            self.threshold = np.zeros(self.base.img.shape[:2], dtype=float)
 
         # Combine the results of a series of images
         for i, img in enumerate(baseline_images):
 
-            # Extract mono-colored version
             probe_img = img.copy()
+
+            # Extract mono-colored version in case of a monochromatic comparison
             self._extract_scalar_information(probe_img)
 
             # Take (unsigned) difference
             diff = skimage.util.compare_images(
                 probe_img.img, self.base.img, method="diff"
             )
 
+            # Extract mono-colored version (should be only active in case of a
+            # multichromatic comparison
+            monochromatic_diff = self._extract_scalar_information_after(diff)
```

Comment:
why not just set it equal to diff? It looks like the same action? is it a memory thing?

Reply:
Again modularity. I found it to be a differnce when the signal is restricted from tri- to monochromatic signals. Either before or after the difference. In order to allow tailoring to any given image, this modularity has been introduced.